crates/nu-command/src/filters/first.rs

```diff
@@ -91,95 +91,84 @@ fn first_helper(
         Some(x) => x as usize,
         None => 1,
     };
 
     let ctrlc = engine_state.ctrlc.clone();
     let metadata = input.metadata();
 
-    let mut input_peek = input.into_iter().peekable();
-    if input_peek.peek().is_some() {
-        match input_peek
-            .peek()
-            .ok_or_else(|| {
-                ShellError::GenericError(
-                    "Error in first".into(),
-                    "unable to pick on next value".into(),
-                    Some(call.head),
-                    None,
-                    Vec::new(),
-                )
-            })?
-            .get_type()
-        {
-            Type::Binary => {
-                match &mut input_peek.next() {
-                    Some(v) => match &v {
-                        Value::Binary { val, .. } => {
-                            let bytes = val;
-                            if bytes.len() >= rows_desired {
-                                // We only want to see a certain amount of the binary
-                                // so let's grab those parts
-                                let output_bytes = bytes[0..rows_desired].to_vec();
-                                Ok(Value::Binary {
-                                    val: output_bytes,
-                                    span: head,
-                                }
-                                .into_pipeline_data())
-                            } else {
-                                // if we want more rows that the current chunk size (8192)
-                                // we must gradually get bigger chunks while testing
-                                // if it's within the requested rows_desired size
-                                let mut bigger: Vec<u8> = vec![];
-                                bigger.extend(bytes);
-                                while bigger.len() < rows_desired {
-                                    match input_peek.next() {
-                                        Some(Value::Binary { val, .. }) => bigger.extend(val),
-                                        _ => {
-                                            // We're at the end of our data so let's break out of this loop
-                                            // and set the rows_desired to the size of our data
-                                            rows_desired = bigger.len();
-                                            break;
-                                        }
-                                    }
-                                }
-                                let output_bytes = bigger[0..rows_desired].to_vec();
-                                Ok(Value::Binary {
-                                    val: output_bytes,
-                                    span: head,
-                                }
-                                .into_pipeline_data())
-                            }
-                        }
+    let input_span = input.span();
```

Comment:
I think the logic can be simplified:
```rust
let input_not_supported_error = || -> ShellError {             ShellError::UnsupportedInput("take does not support this input type".into(), input.span().unwrap_or(call.head)) }
```

Reply:
I intentionally chose slightly different error messages depending on whether we can get the input span.  Not 100% necessary but I think the error messages are a little nicer that way.

Reply:
Oh, sorry for I haven't notice for different error messages.